### PR TITLE
PHR-7 adding  ask to unlock dialog in create proposal button

### PR DESCRIPTION
**Pull request by KolbyML**
This PR basically just asks for the user to enter their password before creating the proposal. This is needed before if a user forgot to unlock the wallet first they would need to fully recreate the proposal.

### CODE REVIEW

File: src/qt/proposallist.cpp [hunk 1]
```diff
@@ -268,45 +268,69 @@ ProposalList::ProposalList(QWidget *parent) :
...
+void ProposalList::setModel(WalletModel* model)
+{
 
-    setLayout(vlayout);
+    QSettings settings;
+    proposalTableModel = new ProposalTableModel(this);
+    this->model = model;
+
+    if (model) {
+        proposalProxyModel = new ProposalFilterProxy(this);
```
Comment (wqking) [inline src/qt/proposallist.cpp:286]: What's the purpose to delay initializing the objects here? Seems they don't need the existence of model. The changes here don't hurt though seems unnecessary.

File: src/qt/proposallist.cpp [hunk 2]
```diff
@@ -268,45 +268,69 @@ ProposalList::ProposalList(QWidget *parent) :
...
+        proposalList->setColumnWidth(ProposalTableModel::VotesNeeded, VOTES_NEEDED_COLUMN_WIDTH);
+
+        columnResizingFixer = new GUIUtil::TableViewLastColumnResizingFixer(proposalList, VOTES_NEEDED_COLUMN_WIDTH, MINIMUM_COLUMN_WIDTH);
+    }
 }
 
 void ProposalList::createProposal()
 {
+
+    WalletModel::EncryptionStatus encStatus = model->getEncryptionStatus();
```
Comment (wqking) [inline src/qt/proposallist.cpp:319]: Though it works, it's better to add nullptr check for `model` before using it.

File: src/qt/proposallist.cpp (hunk 1 above)
Comment (KolbyML) [reply src/qt/proposallist.cpp:286]: This tab was originally based on the transactions tab and when it was being made by someone at chaincoin they removed getModel from the file, this is how the transactions tab originally did it.

File: src/qt/proposallist.cpp (hunk 2 above)
Comment (KolbyML) [reply src/qt/proposallist.cpp:319]: We could probably do that, this is just how it was done in the Masternode page

**Review by wqking — APPROVED** [review]

**Review by tohsnoom — APPROVED** [review]

**Pull request merged**